gsy_myco_sdk/utils.py

```diff
@@ -15,3 +17,37 @@ def websocket_domain_name_from_env():
 
 def simulation_id_from_env():
     return os.environ.get("MYCO_CLIENT_SIMULATION_ID", MYCO_CLIENT_SIMULATION_ID)
+
+
+def log_recommendations_response(data: dict) -> None:
+    """Log the response data of recommendations sent to the clearing mechanism."""
+    recommendations = data["recommendations"]
+    if not recommendations:
+        return
+    logging.info("Length of recommendations: %s", len(recommendations))
+    recommendations_table = []
+    recommendations_table_headers = [
+        "#", "Buyer", "Seller",
+        "Bid kWh", "Offer kWh", "Status", "Message"]
+
+    # Workaround for not printing propagated recommendations
```

Comment:
what are propagated recommendations?

Reply:
I'm talking here about the orders that were forwarded to other markets by an agent

Reply:
Ok, but then I do not get the comment, what is the workaround? 
`uyers / sellers can not be part of multiple recommendations` ?
Can you please explain this a bit more? Thanks!

Reply:
this part of the comment is outdated and misleading, I was looking for ways to distinguish the forwarded orders and originally made the cache by seller and buyer origin, let me remove this part of the comment

Reply:
Sorry, I still not get what the current workaround is.

Reply:
I think I see the workaround here. Hassan is not using the seller_id, but the seller_origin_id and the original_price in the offer_data set. That way even if the offer is forwarded, the key of the set is the same (because it will contain the original offer / bid information, and not the propagated ones). 
This will work, I would personally try to solve this issue without a cache, but by checking whether the seller_id == seller_origin_id to detect if the offer is forwarded or not, however I admit that this approach also works. Hannes is right in that regard that if this approach is maintained a more detailed comment is required because the implementation is quite misleading.

Reply:
@spyrostz exactly!
In case we adopt your proposal, the way I see it there may be some recommendations where this condition `seller_id == seller_origin_id` is True for `buyer_id == buyer_origin_id`.

What's your view on that? should I leave it the way it currently is and add a verbose comment? 
To my mind, adding a unique ID across all forwarded orders is a good idea, but I'm not sure about the drawbacks or the reason it was not previously added

Reply:
@Hassan754 I amended my comment above, the comparison seller_id == seller_origin_id will not work because it might skip some matches (essentially if, e.g. due to grid fees or energy demand, there is match only in a higher market and not in the house market, there is going to be only one recommendation in the grid market, and no recommendation in the house market. In this case we would not log the recommendation at all with my approach, which is obviously incorrect).
Therefore leave it the way it is, it is good to go once you add a verbose comment in order for future us to recall this discussion. Thanks!

Reply:
Ah one final remark, it would be great if, before adding the price to the set, you limited the precision of the floating point numbers to e.g. 5 decimal points (I think that we have a global constant in the gsy-framework for that). Reason for this is because of the casting to string, due to floating point errors it might lead into the cache not being hit correctly, due to slightly different floats. In theory you should not experience this issue because the original price is an assignment and does not get recalculated, therefore the floating point number will not change, however better safe than sorry

Reply:
Thanks for your explanation.
I need to clarify one last question before I fully understand: 
This "issue" only appears when the myco matcher is connected to multiple markets, right? If it only controls one market, this duplication of recommendations is not possible, right?

Reply:
@hannesdiedrich indeed, this is correct

Reply:
Thanks a lot!